### ISSUE TITLE
Labels added to generated YAML seem a bit aggressive

### ISSUE THREAD

**Issue by joelanford** [kubernetes-sigs/kubebuilder#3783]
### What broke? What's expected?

Nothing is broken really, but I find myself removing all (or nearly all) of the `app.kubernetes.io` labels on the YAML the kubebuilder generates.

For example: https://github.com/kubernetes-sigs/kubebuilder/blob/2302f6e131cb6b4a042069bf460f101fcf5e5da3/testdata/project-v4/config/rbac/admiral_editor_role.yaml#L6-L11

I know these are present based on recommendations made here: https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/#labels

I would propose the following:
- `app.kubernetes.io/name` -> Project name
- `app.kubernetes.io/instance` -> Remove; Kubebuilder doesn't know what the instance names will be. This is only known at create time in a cluster.
- `app.kubernetes.io/component` -> Remove; Kubebuilder doesn't know the architecture for this project.
-  `app.kubernetes.io/created-by` -> Remove; Not in the recommended labels doc, also Kubebuilder doesn't know what will eventully create it.
- `app.kubernetes.io/part-of` -> Remove; Kubebuilder doesn't know the architecture for this project.
- `app.kubernetes.io/managed-by` -> Remove; Kubebuilder doesn't know how this project will be managed.

### KubeBuilder (CLI) Version

`master`

**Comment by camilamacedo86** [comment]
Thank you for raising this one.

That makes sense for me. +1. 
I was surprise that we were adding a label not in the docs.

From your list,  just one case scenario I did not go along with. From docs

```
app.kubernetes.io/managed-by | The tool being used to manage the operation of an application Example Helm
```

This label is intended to indicate the tool managing the resource (e.g., Helm/Kustomize and etc).  We scaffold the resources therefore we know the tool used which in our case is only Kustomize for ALL. Right?

**Comment by sarthaksarthak9** [comment]
@joelanford @camilamacedo86 I would love to work on this issue

**Comment by camilamacedo86** [comment]
Closing it is done and released with 3.14.1
